src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -67,6 +69,11 @@ public async Task<IActionResult> Get(string name)
         [Authorize(Policy = PolicyNames.AdminAuthLevel)]
         public async Task<IActionResult> CreateOrUpdate(string name, [FromBody] FunctionMetadataResponse functionMetadata, [FromServices] IFileMonitoringService fileMonitoringService)
         {
+            if (_applicationHostOptions.Value.IsFileSystemReadOnly)
+            {
+                return BadRequest("Filesystem is read-only");
```

Comment:
nit: more descriptive error message - `Your app is currently in read only mode. Cannot create or update functions`.
For reference - Portal displays - `Your app is currently in read only mode because you are running from a package file. To make any changes update the content in your zip file and WEBSITE_RUN_FROM_PACKAGE app setting.` when function app content cannot be edited.